### DISCUSSION TITLE
Random desk movements

### DISCUSSION THREAD

**Discussion by chrishas35** [tjhorner/wifi-desk-controller#12] [Support] [answered]
Well, this has now happened twice... without interaction on my part, the desk has moved 0.1 and then  0.2.  You can seem the movements on the HA graph below:

![image](https://user-images.githubusercontent.com/35075/175108354-712a63fb-cc50-4796-8b32-5ebea2e91a53.png)

There doesn't seem to be a consistency in the time in which the movements occur. After the first two movements you can see that I moved it back up to the preset... I'm going to leave it alone this time and see if it moves again.

There does seem to be a probable correlation to the target height going unavailable, however, it coming back is at the same height it as previously... and the actual height is now 32.1 after the two movements.

![image](https://user-images.githubusercontent.com/35075/175109182-a5174608-19ad-40a6-95b6-3405865eec57.png)

**Comment by chrishas35** [comment]
So I suspect this is happening because the wifi desk controller is going offline or something, and it's doing the slight movement mentioned in the instructions about first connection.  I didn't make that connection because initially my desk was moving all the way down but as discussed in #4 it wouldn't connect, and when I did the workaround it did not move at all. I currently have USB power disconnected...I'm going to redo and leave USB power in place and see if it'll hold.

**Reply by chrishas35** [reply]
Since reconnecting and leaving USB power plugged in, I have not seen this behavior again.

**Reply by tjhorner** [reply]
It definitely does sound like the board may be losing power, perhaps because it's drawing too much current from the desk control box. If it was a software reset, then the desk wouldn't move; that only happens if it's a hardware reset.

I will start recommending that people keep the board powered from micro USB, since it seems to alleviate both this issue and #6.

Thanks for the report, and for trying out the workaround — this is something that I will fix in v2.

**Reply by tjhorner** [reply]
Also, something you can do to verify that it is indeed a hardware reset is adding a [debug component](https://esphome.io/components/debug.html) to your ESPHome config. Make sure to set the log level to debug as well.

It will output the reset reason on restart. Should this happen again, check the ESPHome logs for details.

**Comment by tjhorner** [comment, marked as answer]
To work around this issue, follow the steps [listed in the wiki](https://github.com/tjhorner/wifi-desk-controller/wiki/Troubleshooting#my-desk-occasionally-lowers-or-raises-slightly).
